global.php

```diff
@@ -745,8 +745,11 @@
 	}
 }
 
-// Get awaiting moderation queue stats
-if($can_access_moderationqueue || ($mybb->user['ismoderator'] && $mybb->usergroup['canmodcp'] == 1 && $mybb->usergroup['canmanagemodqueue'] == 1))
+// Get awaiting moderation queue stats, except if the page is editpost.php,
+// because that page can make changes - (un)approving attachments, or deleting
+// unapproved attachments - that would invalidate anything generated here.
+// Just leave this queue notification blank for editpost.php.
+if(!(defined('THIS_SCRIPT') && THIS_SCRIPT == 'editpost.php') && ($can_access_moderationqueue || ($mybb->user['ismoderator'] && $mybb->usergroup['canmodcp'] == 1 && $mybb->usergroup['canmanagemodqueue'] == 1)))
```

Comment:
This will simply hide the notification in the editpost page, is that correct?

Reply:
Correct. It hides _all_ moderation queue notices though, so on top of unapproved attachment notices, it also hides unapproved thread and unapproved post notices. If we wanted to be more fine-grained, and only hide unapproved _attachment_ notices for the editpost page, then we could instead conditionally alter the $modqueue_types array a few lines below (i.e., remove 'attachments' from it when THIS_SCRIPT == 'editpage.php').

Reply:
I'm fine with this, but lets hear @mybb/developers

Reply:
I don't think it's a wise move of making the header visually inconsistent across forum modules.

Reply:
OK, @yuliu, but if we do nothing then the header will also be visually inconsistent across forum modules - it will at times for the editpage.php module alone show an unapproved attachment in the queue when there are none, and fail to show an unapproved attachment in the queue when there is one - so, we're kind of "damned if we do; damned if we don't".

Question for you: I've tried three different approaches to this issue (1. regenerate the entire header; 2. selectively replace only the mod queue in the header; 3. selectively disable the mod queue in the header for editpage.php), and you've rejected all three of them, so: what do you think _should_ be done; what is it that you _would_ accept?

Reply:
I myself would be fine with not fixing this right now, if this can be fixed by moving the header and footer build process to functions then this could as well be fixed then.

Reply:
OK. Thank you, @Sama34. I will wait for a little while for others to provide their feedback, but, unless/until a contrary view is expressed, I will assume that we are going to go with the following approach: ignore this potential UI inconsistency in the header for now; don't try to fix it at all or in any way - we will leave any such fix until 1.9.*. @lairdshaw  should revert the (by now minor) changes that he has made in this PR (to global.php) to try to minimise the effects of the UI inconsistency.

Yes? No? Anybody want to express a contrary view?

Reply:
> but if we do nothing then the header will also be visually inconsistent across forum modules

Yes and such inconsistency may also appear across browser windows / sessions and that's something we're probably gonna to ignore.

> so, we're kind of "damned if we do; damned if we don't".

Exactly.

> Question for you: I've tried three different approaches to this issue (1. regenerate the entire header; 2. selectively replace only the mod queue in the header; 3. selectively disable the mod queue in the header for editpage.php), and you've rejected all three of them, so: what do you think _should_ be done; what is it that you _would_ accept?

Leave current problem (the volatile count number of unapproved attachments) as is. The reasons are: 
- editpost page is a screen focused on editing a post, users probably won't navigate to other pages without saving the post (means exiting editpost page) or will dismiss the changes made (means exiting editpost page).
- no more compatible way to redo the work in _global.php_, proposals by @lairdshaw may break more compared with fixing this one issue, especially on heavily customized forums (by custom codes and/or plugins).

So I suggested to open a discussion about re-design how a page's header/footer/etc. are evaluated which will probably resolve the problem in this PR and the common need of editing relevant templates by plugins.

Reply:
This one seems OK to me now after reading how the pm_notice is taken care of on https://github.com/mybb/mybb/blob/mybb_1822/global.php#L846 , looks like it's not a real solution but only a workaround for now.

Furthermore, the unread pm count has the same issue as yours, see https://github.com/mybb/mybb/blob/mybb_1822/global.php#L547 .

Reply:
Interesting, @yuliu. So, there's a precedent for this approach in the code already. Thanks for digging it up.

Reply:
Above, I wrote:

> unless/until a contrary view is expressed, I will assume that [...] @lairdshaw should revert the (by now minor) changes that he has made in this PR (to global.php) to try to minimise the effects of the UI inconsistency.

I took @yuliu's later approving comments to be the expression of a contrary view, and so I have not reverted those changes after all. At this point, a couple of days later, I think that unless anybody wants to argue otherwise, this PR is ready as-is.